src/main/java/io/stargate/sgv2/jsonapi/service/operation/filters/table/codecs/JSONCodec.java

```diff
@@ -0,0 +1,135 @@
+    implements BiPredicate<DataType, Object> {
+
+  /**
+   * Call to check if this codec can convert the type of the `value` into the type needed for a
+   * column of the `targetCQLType`.
+   *
+   * <p>Used to filter the list of codecs to find one that works, which can then be unchecked cast
+   * using {@link JSONCodec#unchecked(JSONCodec)}
+   *
+   * @param targetCQLType {@link DataType} of the CQL column the value will be written to.
+   * @param value Instance of a Java value that will be written to the column.
+   * @return True if the codec can convert the value into the type needed for the column.
+   */
+  @Override
+  public boolean test(DataType targetCQLType, Object value) {
+    // java value tests comes from TypeCodec.accepts(Object value) in the driver
+    return this.targetCQLType.equals(targetCQLType)
+        && javaType.getRawType().isAssignableFrom(value.getClass());
+  }
+
+  /**
+   * Applies the codec to the value.
+   *
+   * @param value Json value of type {@link JavaT} that needs to be transformed into the type CQL
+   *     expects.
+   * @return Value of type {@link CqlT} that the CQL driver expects.
+   * @throws FromJavaCodecException if there was an error converting the value.
+   */
+  public CqlT apply(JavaT value) throws FromJavaCodecException {
+    return fromJava.apply(value, targetCQLType);
+  }
+
+  @SuppressWarnings("unchecked")
+  public static <JavaT, CqlT> JSONCodec<JavaT, CqlT> unchecked(JSONCodec<?, ?> codec) {
+    return (JSONCodec<JavaT, CqlT>) codec;
+  }
+
+  /**
+   * Function interface that is used by the codec to convert the Java value to the value CQL
+   * expects.
+   *
+   * <p>The interface is used so the conversation function can throw the checked {@link
+   * FromJavaCodecException} and the function is also passed the target type so it can construct a
+   * better exception.
+   *
+   * <p>Use the static constructors on the interface to get instances, see it's use in the {@link
+   * JSONCodecRegistry}
+   *
+   * @param <T> The type of the Java object that needs to be transformed into the type CQL expects
+   * @param <R> The type Java object the CQL driver expects
+   */
+  @FunctionalInterface
+  public interface FromJava<T, R> {
+
+    /**
+     * Convers the current Java value to the type CQL expects.
+     *
+     * @param t
+     * @param targetType The type of the CQL column the value will be written to, passed so it can
+     *     be used when creating an exception if there was a error doing the transformation.
+     * @return
+     * @throws FromJavaCodecException
+     */
+    R apply(T t, DataType targetType) throws FromJavaCodecException;
+
+    /**
+     * Returns an instance that just returns the value passed in, the same as {@link
+     * Function#identity()}
+     *
+     * <p>Unsafe because it does not catch any errors from the conversion, because there are none.
+     *
+     * @return
+     * @param <T>
+     */
+    static <T> FromJava<T, T> unsafeIdentity() {
```

Comment:
And... what is the point here? Is it for type-casting purpose or why is this needed?

Reply:
added a TODO comment